src/v/kafka/server/handlers/list_partition_reassignments.cc

```diff
@@ -0,0 +1,183 @@
+                  reassignments.removing_replicas.push_back(bshard.node_id);
+              }
+          });
+    }
+
+    return reassignments;
+}
+
+// Returns all reassignments currently in progress
+std::vector<ongoing_topic_reassignment> all_active_partition_reassignments(
+  const cluster::topic_table::updates_t& in_progress) {
+    // The in progress reassignments are in a map: ntp -> list of brokers which
+    // is not suitable to use with the ListPartitionReassignments response. So
+    // we restructure them here.
+    std::unordered_map<model::topic, ongoing_topic_reassignment>
+      reassignments_by_topic;
+
+    for (auto& [ntp, status] : in_progress) {
+        // Convert list of broker shards to node ids
+        std::vector<model::node_id> replicas;
+        std::for_each(
+          status.get_previous_replicas().begin(),
+          status.get_previous_replicas().end(),
+          [&replicas](const model::broker_shard& bshard) {
+              replicas.push_back(bshard.node_id);
+          });
+
+        if (reassignments_by_topic.contains(ntp.tp.topic)) {
+            reassignments_by_topic[ntp.tp.topic].partitions.push_back(
+              ongoing_partition_reassignment{
+                .partition_index = ntp.tp.partition, .replicas = replicas});
+        } else {
+            ongoing_topic_reassignment topic_reassignment{.name = ntp.tp.topic};
+            topic_reassignment.partitions.push_back(
+              ongoing_partition_reassignment{
+                .partition_index = ntp.tp.partition, .replicas = replicas});
+            reassignments_by_topic[ntp.tp.topic] = topic_reassignment;
+        }
+    }
+
+    std::vector<ongoing_topic_reassignment> all_active_reassignments;
+    for (auto& [_, reassignments] : reassignments_by_topic) {
+        all_active_reassignments.push_back(reassignments);
+    }
+
+    return all_active_reassignments;
+}
+
+template<>
+ss::future<response_ptr> list_partition_reassignments_handler::handle(
+  request_context ctx, [[maybe_unused]] ss::smp_service_group ssg) {
+    list_partition_reassignments_request request;
+    request.decode(ctx.reader(), ctx.header().version);
+    log_request(ctx.header(), request);
+    list_partition_reassignments_response resp;
+
+    if (!ctx.authorized(
+          security::acl_operation::describe, security::default_cluster_name)) {
+        resp.data.error_code = error_code::cluster_authorization_failed;
+        resp.data.error_message = ss::sstring{
+          error_code_to_str(error_code::cluster_authorization_failed)};
+        co_return co_await ctx.respond(std::move(resp));
+    }
+
+    // TODO(@NyaliaLui) - Check if we are the controller
+
+    // If topics is undefined, get all active reassignments
+    if (!request.data.topics.has_value()) {
+        auto all_active_reassignments = all_active_partition_reassignments(
+          ctx.metadata_cache().updates_in_progress());
```

Comment:
are all of the updates really mirrored across all the cores so that we can find all the updates on any core serving the request?

Reply:
Good point, if this is the case you will need to lookup which core is responsible for the reassignment before querying that collection

Reply:
At this point in the code the `topics` vector is undefined so I cannot create a `ntp` and get the shard for a specific reassignment/request. Instead I could:
1. Get the active reassignments of the core that is handling the request with `ss::this_shard_id` or
2. Get the active reassignments on every core and then merge them into a single container on the core handling the request. But is this design ideal? For some reason I recall that we try to reduce as much cross-core traffic as possible.

Reply:
>I could get the active reassignments on every core and then merge them into a single container on the core handling the request. But is this design ideal? For some reason I recall that we try to avoid cross-core traffic.

are you alluding to the core assignment metadata being partitioned not duplicated on every core?

Reply:
> > I could get the active reassignments on every core and then merge them into a single container on the core handling the request. But is this design ideal? For some reason I recall that we try to avoid cross-core traffic.
> 
> are you alluding to the core assignment metadata being partitioned not duplicated on every core?

Yes.

Reply:
im confused then. we want to return active reassignments to the user, but how do we know those are registered on the current core? i was thinking that all the state might be replicated on all the cores so these answers could be answered on any core. that is to me the question about what is happening here.

Reply:
Looks like Michal answered this Q for me. Looks like the state is replicated on every core so accessing via `this shard` should be OK.
https://github.com/redpanda-data/redpanda/pull/7685#discussion_r1081045802